File: Solution in CPP/Que12.cpp
// Find the maximum subarray sum in Rust

#include<bits/stdc++.h>
using namespace std;

int main(){
    vector<int> array = {1,4,5,3,-8,1,5,7,-1,-2,8};
    
    int maxSubarraySum = INT_MIN;
    int maxSum = 0;
    for(int i= 0; i<array.size(); i++){
        maxSum += array[i];
        if(maxSum > maxSubarraySum){
            maxSubarraySum = maxSum;
        }

        if(maxSum <= 0){
            maxSum = 0;
        }
    }

    cout<<"Answer = "<<maxSubarraySum<<endl;
    return 0;
}

File: Solution in RUST/Que12.rs
// Find the maximum subarray sum in Rust
fn main() {
    let array = vec![1, 4, 5, 3, -8, 1, 5, 7, -1, -2, 8];

    let mut max_subarray_sum = i32::MIN;
    let mut max_sum = 0;
    for &num in &array {
        max_sum += num;
        if max_sum > max_subarray_sum {
            max_subarray_sum = max_sum;
        }

        if max_sum <= 0 {
            max_sum = 0;
        }
    }

    println!("Answer = {}", max_subarray_sum);
}
